osu.Game/Rulesets/LegacyBeatmapConversionDifficultyInfo.cs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using osu.Game.Beatmaps;
+using osu.Game.Online.API.Requests.Responses;
+using osu.Game.Rulesets.Objects.Types;
+
+namespace osu.Game.Rulesets
+{
+    /// <summary>
+    /// A set of properties that are required to facilitate beatmap conversion between legacy rulesets.
+    /// </summary>
+    public class LegacyBeatmapConversionDifficultyInfo : IBeatmapDifficultyInfo
+    {
+        /// <summary>
+        /// Whether the beatmap is made for the target ruleset.
+        /// </summary>
+        public bool IsForTargetRuleset { get; set; }
+
+        /// <summary>
+        /// The beatmap circle size.
+        /// </summary>
+        /// <remarks>
+        /// Only used if <see cref="IsForTargetRuleset"/> is <c>false</c>.
+        /// </remarks>
+        public float CircleSize { get; set; }
+
+        /// <summary>
+        /// The beatmap overall difficulty.
+        /// </summary>
+        /// <remarks>
+        /// Only used if <see cref="IsForTargetRuleset"/> is <c>false</c>.
+        /// </remarks>
+        public float OverallDifficulty { get; set; }
+
+        /// <summary>
+        /// The count of hitcircles in the beatmap.
+        /// </summary>
+        /// <remarks>
+        /// Only used if <see cref="IsForTargetRuleset"/> is <c>false</c>.
+        /// <para>When converting from osu! ruleset beatmaps, this is equivalent to the sum of sliders and spinners in the beatmap.</para>
```

Comment:
Looking across usages, this only appears to be the case for mania, and only when going through the weird `FromBeatmap()` method below.